Azure-Samples/openhack-devops-team
# POI - Increased Code Coverage

## Purpose
This PR adds extra unit tests to POI to provide code coverage. Note: This is focused on Unit Testing and not integration testing coverage so Program.cs and Startup.cs were excluded from Code Coverage.

This PR brings unit test code coverage to 95.65% for POI.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe:
Unit Tests for Code Coverage
```

## How to Test
*  Get the code
* ``` cd apis/poi ```
* ``` dotnet test ./tests/UnitTests ```
* All tests should pass

To run code coverage using coverlet / report generator
Install tools:
 ```bash
dotnet tool install -g dotnet-reportgenerator-globaltool 
dotnet add package coverlet.msbuild 
```
Run tests and collect coverage:
```bash
dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura 
reportgenerator -reports:coverage.cobertura.xml -report-types:Html -targetdir:Report 
```

## What to Check
* Unit Tests should pass
* Code Coverage should be higher than 80%

## Other Information
No changes were made to the core api code, other than adding thhe ExcludeFromCodeCoverage annotation. In a future PR we can remove that attribute with increased integration tests.
